cmd/fleet/server.go

```diff
@@ -105,11 +105,11 @@ func runServer(ctx context.Context, router *httprouter.Router, cfg *config.Serve
 		ln = wrapConnLimitter(ctx, ln, cfg)
 
 		if cfg.TLS != nil && cfg.TLS.IsEnabled() {
-			commonTlsCfg, err := tlscommon.LoadTLSServerConfig(cfg.TLS)
+			commonTLSCfg, err := tlscommon.LoadTLSServerConfig(cfg.TLS)
 			if err != nil {
 				return err
 			}
-			server.TLSConfig = commonTlsCfg.ToConfig()
+			server.TLSConfig = commonTLSCfg.BuildServerConfig(cfg.Host)
```

Comment:
https://github.com/elastic/fleet-server/blob/4e3c8b0e571e96921ab7c69c30ab8a857a4690d3/cmd/fleet/server.go#L106-L118

This is now the same as main so 👍 

```go
			commonTLSCfg, err := tlscommon.LoadTLSServerConfig(cfg.TLS)
			if err != nil {
				return err
			}
			server.TLSConfig = commonTLSCfg.BuildServerConfig(cfg.Host)

			// Must enable http/2 in the configuration explicitly.
			// (see https://golang.org/pkg/net/http/#Server.Serve)
			server.TLSConfig.NextProtos = []string{"h2", "http/1.1"}

			ln = tls.NewListener(ln, server.TLSConfig)
```

Reply:
`BuildServerConfig` function was introduced in beats 7.12.x when we updated the required version of go to 1.15.x
It specifies a custom `VerifyConnection` attribute that is need for the changes with certificate verification introduced in that version of go